virtual_rainforest/models/soil/carbon.py

```diff
@@ -0,0 +1,203 @@
+        if any(i < 0 for i in maom) or any(i < 0 for i in lmwc):
+            log_and_raise(
+                "Initial carbon pools contain at least one negative value!",
+                InitialisationError,
+            )
+
+        self.maom = maom
+        self.lmwc = lmwc
+
+    def update_pools(
+        self,
+        pH: NDArray[np.float32],
+        bulk_density: NDArray[np.float32],
+        soil_moisture: NDArray[np.float32],
+        soil_temp: NDArray[np.float32],
+        percent_clay: NDArray[np.float32],
+        dt: float,
+    ) -> None:
+        """Update all soil carbon pools.
+
+        This function calls lower level functions which calculate the transfers between
+        pools. When all transfers have been calculated the net transfer is used to
+        update the soil pools.
+
+        Args:
+            pH: pH values for each soil grid cell
+            bulk_density: bulk density values for each soil grid cell
+            soil_moisture: soil moisture for each soil grid cell
+            soil_temp: soil temperature for each soil grid cell
+            percent_clay: Percentage clay for each soil grid cell
+            dt: time step (days)
+        """
+        # TODO - Add interactions which involve the three missing carbon pools
+
+        lmwc_to_maom = self.mineral_association(
+            pH, bulk_density, soil_moisture, soil_temp, percent_clay
+        )
+
+        # Once changes are determined update all pools
+        self.lmwc -= lmwc_to_maom * dt
+        self.maom += lmwc_to_maom * dt
+
+    def mineral_association(
+        self,
+        pH: NDArray[np.float32],
+        bulk_density: NDArray[np.float32],
+        soil_moisture: NDArray[np.float32],
+        soil_temp: NDArray[np.float32],
+        percent_clay: NDArray[np.float32],
+    ) -> NDArray[np.float32]:
+        """Calculates net rate of LMWC association with soil minerals.
+
+        Following Abramoff et al. (2018), mineral adsorption of carbon is controlled by
+        a Langmuir saturation function. At present, binding affinity and Q_max are
+        recalculated on every function called based on pH, bulk density and clay
+        content. Once a decision has been reached as to how fast pH and bulk density
+        will change (if at all), this calculation may need to be moved elsewhere.
+
+        Args:
+            pH: pH values for each soil grid cell
+            bulk_density: bulk density values for each soil grid cell
+            soil_moisture: soil moisture for each soil grid cell
+            soil_temp: soil temperature for each soil grid cell
+            percent_clay: Percentage clay for each soil grid cell
+
+        Returns:
+            lmwc_to_maom: The net flux from LMWC to MAOM
+        """
+
+        # This expression is drawn from (Mayes et al. (2012))
+        binding_affinity = 10.0 ** (
+            BINDING_WITH_PH["slope"] * pH + BINDING_WITH_PH["intercept"]
+        )
```

Comment:
Hah - was going to suggest `np.power` here, but turns out be no better:

```python
In [18]: import timeit
In [19]: timeit.timeit("10.0 ** np.arange(10)", setup="import numpy as np")
Out[19]: 1.8152062078006566
In [23]: timeit.timeit("np.power(10.0, np.arange(10))", setup="import numpy as np")
Out[23]: 1.8450454170815647
```

And should you accidentally set that up with an integer? 
```
In [20]: timeit.timeit("np.power(10, np.arange(10))", setup="import numpy as np")
Out[20]: 3.010193958878517
```